src/CoreWCF.RabbitMQ/src/CoreWCF/Channels/RabbitMqConnectionSettings.cs

```diff
@@ -0,0 +1,177 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Linq;
+using System.Net;
+using RabbitMQ.Client;
+
+namespace CoreWCF.Channels
+{
+    public class RabbitMqConnectionSettings
+    {
+        public const string TempQueuePrefix = "corewcf-temp-";
+        private const string AMQPScheme = "amqp";
+        private const string SecureAMQPScheme = "amqps";
+        private const string DefaultVirtualHost = "/";
+
+        public string Host { get; set; }
+        public int Port { get; set; }
+        public string Exchange { get; set; } = string.Empty;
+        public string QueueName { get; set; }
+        public string VirtualHost { get; set; } = DefaultVirtualHost;
+        public virtual string RoutingKey { get; set; }
+        public string UserName { get; set; }
+        public string Password { get; set; }
+        public SslOption SslOption { get; set; }
+        public bool AutomaticRecoveryEnabled => true;
+
+        public static RabbitMqConnectionSettings FromUri(
+            Uri uri,
+            ICredentials credentials = null,
+            SslOption sslOption = null,
+            string virtualHost = DefaultVirtualHost)
+        {
+            sslOption = ConfigureSslOption(sslOption, uri);
+            var queueName = GetQueueNameFromUri(uri);
```

Comment:
Need a null check on uri otherwise ConfigureSslOption can throw NRE if sslOption is null.